tasks/eva_3380/src/main/groovy/eva3380/eva3380_collect_colliding_ss_hashes.groovy

```diff
@@ -0,0 +1,168 @@
+                            oldSVE.clusteredVariantAccession, oldSVE.isSupportedByEvidence(),
+                            oldSVE.isAssemblyMatch(), oldSVE.isAllelesMatch(), oldSVE.isValidated(),
+                            oldSVE.createdDate)
+                    def updatedHash = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction()).apply(updatedSV)
+                    // Crudely tack on the old SS hash to the start of the remappingId since that is the
+                    // least harmless String attribute that allows us to encode extra information
+                    // without messing up the object model's integrity
+                    // We must be careful with this however when using this info in EVA-3380 to update Indels
+                    return new SubmittedVariantEntity(oldSVE.accession, updatedHash, updatedSV, oldSVE.version,
+                            oldSVE.remappedFrom, oldSVE.remappedDate,
+                            oldSVE.hashedMessage + "_" + oldSVE.remappingId)
+                }
+            }
+    return normalizedDbsnpAndEvaSves
+}
+
+def getClashingSSRecordsFromDb = {List<SubmittedVariantEntity> svesFromVCF,
+                                  EVADatabaseEnvironment devEnv, EVADatabaseEnvironment prodEnv ->
+    def svesFromVcfGroupedByHash = svesFromVCF.groupBy{it.hashedMessage}
+    def clashingSvesInDevAndProd = [devEnv, prodEnv].collect{env ->
+        [sveClass, dbsnpSveClass].collect{collectionClass ->
+            env.mongoTemplate.find(query(where("_id").in(svesFromVcfGroupedByHash.keySet())), collectionClass)
+    }.flatten()}
+    def clashingSSAccessionsInProd = clashingSvesInDevAndProd[1].collect{it.accession}.toSet()
+    def prodSSAccessionsInNormalizedPopulation =
+            [sveClass, dbsnpSveClass].collect{collectionClass ->
+                devEnv.mongoTemplate.find(query(where("accession").in(clashingSSAccessionsInProd).and("seq").is(
+                        svesFromVCF.first().referenceSequenceAccession)), collectionClass)
+            }.flatten().collect{it.accession}.toSet()
+    // Don't include clashing SS, that are also lined up for normalization, as a merge candidate
+    // See here for illustration: https://docs.google.com/spreadsheets/d/13go87r2jR__iMS5YxDmJ55TAFEwMD0COZOjKnd-0934/edit#rangeid=122505042
+    // This way we know for sure that the SS mergeTarget/mergee won't change when another SS normalization is processed
+    clashingSvesInDevAndProd[1].removeIf{prodSSAccessionsInNormalizedPopulation.contains(it.accession)}
+    def clashingSvesGroupedByHash =
+            clashingSvesInDevAndProd.flatten().toUnique{"" + "${it.hashedMessage}_${it.accession}"}.groupBy{it.hashedMessage}
+    def hashesFromClashingSSCollection =
+            devEnv.mongoTemplate.find(query(where("_id").in(svesFromVcfGroupedByHash.keySet())),
+                    ClashingSSHashes.class).groupBy{it.ssHash}
```

Comment:
Can you use `collectEntries` instead of `groupBy` since we know there should be only one `ClashingSSHashes` per hash?

Reply:
Addressed